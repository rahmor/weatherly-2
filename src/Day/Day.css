.Day {
  border-radius: 8px;
  box-shadow: 0px 0px 5px -2px black, -0px -0px 5px -10px black;
  margin: 8px 0 8px 0;
  background-color: rgba(119, 83, 172, 0.25);
  border: 1px solid white;
  color: white;
}
