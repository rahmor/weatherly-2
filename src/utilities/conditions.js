const CONDITIONS = {
  '01d': {
    color: 'white',
    image: 'clear-sky-paris',
    icon: '01d',
    large: '1vZfSWJ',
    small: '4N6HxcD',
  },
  '02d': {
    color: 'yellow',
    image: 'few-clouds-grass',
    icon: '02d',
    large: 'QQTjqk2',
    small: 'TLWPB94',
  },
  '03d': {
    color: 'gold',
    image: 'scattered-clouds-sun',
    icon: '03d',
    large: 'Vp0Sn2X',
    small: '1fwLqq7',
  },
  '04d': {
    color: 'orange',
    image: 'broken-clouds-sunset',
    icon: '04d',
    large: '5MDn8BX',
    small: '0rgH8Px',
  },
  '09d': {
    color: 'peach',
    image: 'rain-shower-london',
    icon: '90d',
    large: 'BtDhZn4',
    small: 'RpwYNzV',
  },
  '10d': {
    color: 'blue',
    image: 'rain-ground',
    icon: '10d',
    large: 'Qd6ZmVj',
    small: 'D4Vvt0K',
  },
  '11d': {
    color: 'purple',
    image: 'thunder-storm-sydney',
    icon: '11d',
    large: 'sCwJZM9',
    small: 'DMH6W6g',
  },
  '13d': {
    color: 'light-purple',
    image: 'snow-space-needle',
    icon: '13d',
    large: 'mC2jdW1',
    small: 'fv3WGSv',
  },
  '50d': {
    color: 'aquamarine',
    image: 'mist-golden-gate',
    icon: '50d',
    large: 'QmCmQMR',
    small: 'N66BLVC',
  },
  '01n': {
    color: 'white',
    image: 'clear-sky-paris',
    icon: '01n',
    large: '1vZfSWJ',
    small: '4N6HxcD',
  },
  '02n': {
    color: 'yellow',
    image: 'few-clouds-grass',
    icon: '02n',
    large: 'QQTjqk2',
    small: 'TLWPB94',
  },
  '03n': {
    color: 'gold',
    image: 'scattered-clouds-sun',
    icon: '03n',
    large: 'Vp0Sn2X',
    small: '1fwLqq7',
  },
  '04n': {
    color: 'orange',
    image: 'broken-clouds-sunset',
    icon: '04n',
    large: '5MDn8BX',
    small: '0rgH8Px',
  },
  '09n': {
    color: 'peach',
    image: 'rain-shower-london',
    icon: '90n',
    large: 'BtDhZn4',
    small: 'RpwYNzV',
  },
  '10n': {
    color: 'blue',
    image: 'rain-ground',
    icon: '10n',
    large: 'Qd6ZmVj',
    small: 'D4Vvt0K',
  },
  '11n': {
    color: 'purple',
    image: 'thunder-storm-sydney',
    icon: '11n',
    large: 'sCwJZM9',
    small: 'DMH6W6g',
  },
  '13n': {
    color: 'light-purple',
    image: 'snow-space-needle',
    icon: '13n',
    large: 'mC2jdW1',
    small: 'fv3WGSv',
  },
  '50n': {
    color: 'aquamarine',
    image: 'mist-golden-gate',
    icon: '50n',
    large: 'QmCmQMR',
    small: 'N66BLVC',
  },
};

export default CONDITIONS;
