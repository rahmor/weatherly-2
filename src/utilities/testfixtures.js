import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunkMiddleWare from 'redux-thunk';
import reducers from '../reducers/reducers';

export const initialState = {
  fetchError: false,
  city: 'Washington, DC',
  fetching: false,
  current: {
    temp: 74,
    time: '12:41:09 PM',
    dt: 1605739065,
    hi: 74,
    lo: 55,
    icon: '01d',
    condition: 'clear sky',
  },
  daily: [
    {
      dt: 1604851200,
      sunrise: 1604835818,
      sunset: 1604872819,
      temp: {
        day: 74.52,
        min: 55.87,
        max: 74.52,
        night: 61.5,
        eve: 72.34,
        morn: 56.86,
      },
      feels_like: {
        day: 73.72,
        night: 61.57,
        eve: 72.37,
        morn: 54.99,
      },
      pressure: 1028,
      humidity: 46,
      dew_point: 52.39,
      wind_speed: 2.73,
      wind_deg: 270,
      weather: [
        {
          id: 800,
          main: 'Clear',
          description: 'clear sky',
          icon: '01d',
        },
      ],
      clouds: 1,
      pop: 0,
      uvi: 2.98,
    },
    {
      dt: 1604937600,
      sunrise: 1604922286,
      sunset: 1604959164,
      temp: {
        day: 66.88,
        min: 57.09,
        max: 73.11,
        night: 60.24,
        eve: 71.01,
        morn: 58.06,
      },
      feels_like: {
        day: 67.14,
        night: 57.18,
        eve: 70.03,
        morn: 58.51,
      },
      pressure: 1028,
      humidity: 67,
      dew_point: 55.85,
      wind_speed: 2.64,
      wind_deg: 197,
      weather: [
        {
          id: 800,
          main: 'Clear',
          description: 'clear sky',
          icon: '01d',
        },
      ],
      clouds: 0,
      pop: 0,
      uvi: 3.12,
    },
    {
      dt: 1605024000,
      sunrise: 1605008753,
      sunset: 1605045510,
      temp: {
        day: 66.69,
        min: 57.88,
        max: 73.29,
        night: 65.91,
        eve: 72.18,
        morn: 57.99,
      },
      feels_like: {
        day: 66.24,
        night: 65.01,
        eve: 70.9,
        morn: 56.28,
      },
      pressure: 1023,
      humidity: 77,
      dew_point: 59.58,
      wind_speed: 6.11,
      wind_deg: 187,
      weather: [
        {
          id: 802,
          main: 'Clouds',
          description: 'scattered clouds',
          icon: '03d',
        },
      ],
      clouds: 25,
      pop: 0,
      uvi: 3.18,
    },
    {
      dt: 1605110400,
      sunrise: 1605095221,
      sunset: 1605131858,
      temp: {
        day: 71.44,
        min: 65.07,
        max: 74.68,
        night: 69.66,
        eve: 73.02,
        morn: 65.07,
      },
      feels_like: {
        day: 70.34,
        night: 72.66,
        eve: 75.51,
        morn: 65.75,
      },
      pressure: 1017,
      humidity: 79,
      dew_point: 64.62,
      wind_speed: 11.01,
      wind_deg: 199,
      weather: [
        {
          id: 502,
          main: 'Rain',
          description: 'heavy intensity rain',
          icon: '10d',
        },
      ],
      clouds: 94,
      pop: 1,
      rain: 18.99,
      uvi: 3.06,
    },
    {
      dt: 1605196800,
      sunrise: 1605181688,
      sunset: 1605218208,
      temp: {
        day: 67.6,
        min: 59.16,
        max: 69.44,
        night: 59.16,
        eve: 65.82,
        morn: 66.15,
      },
      feels_like: {
        day: 67.59,
        night: 54.82,
        eve: 63.1,
        morn: 67.84,
      },
      pressure: 1013,
      humidity: 84,
      dew_point: 62.8,
      wind_speed: 7.63,
      wind_deg: 346,
      weather: [
        {
          id: 503,
          main: 'Rain',
          description: 'very heavy rain',
          icon: '10d',
        },
      ],
      clouds: 100,
      pop: 1,
      rain: 55.47,
      uvi: 2.87,
    },
    {
      dt: 1605283200,
      sunrise: 1605268156,
      sunset: 1605304559,
      temp: {
        day: 57.94,
        min: 52.02,
        max: 61.21,
        night: 52.02,
        eve: 59.22,
        morn: 56.07,
      },
      feels_like: {
        day: 51.24,
        night: 45.32,
        eve: 53.35,
        morn: 50.29,
      },
      pressure: 1020,
      humidity: 49,
      dew_point: 39.11,
      wind_speed: 7.58,
      wind_deg: 359,
      weather: [
        {
          id: 803,
          main: 'Clouds',
          description: 'broken clouds',
          icon: '04d',
        },
      ],
      clouds: 65,
      pop: 0.05,
      uvi: 2.95,
    },
    {
      dt: 1605369600,
      sunrise: 1605354623,
      sunset: 1605390912,
      temp: {
        day: 52.93,
        min: 46.72,
        max: 57.54,
        night: 54.54,
        eve: 56.77,
        morn: 48.25,
      },
      feels_like: {
        day: 47.39,
        night: 48.36,
        eve: 51.55,
        morn: 42.01,
      },
      pressure: 1027,
      humidity: 54,
      dew_point: 36.77,
      wind_speed: 4.85,
      wind_deg: 77,
      weather: [
        {
          id: 501,
          main: 'Rain',
          description: 'moderate rain',
          icon: '10d',
        },
      ],
      clouds: 73,
      pop: 1,
      rain: 7.01,
      uvi: 2.88,
    },
    {
      dt: 1605456000,
      sunrise: 1605441090,
      sunset: 1605477267,
      temp: {
        day: 62.55,
        min: 56.44,
        max: 67.84,
        night: 60.42,
        eve: 66.51,
        morn: 56.97,
      },
      feels_like: {
        day: 62.78,
        night: 60.3,
        eve: 67.44,
        morn: 56.16,
      },
      pressure: 1013,
      humidity: 87,
      dew_point: 58.78,
      wind_speed: 4.5,
      wind_deg: 268,
      weather: [
        {
          id: 502,
          main: 'Rain',
          description: 'heavy intensity rain',
          icon: '10d',
        },
      ],
      clouds: 100,
      pop: 1,
      rain: 27.04,
      uvi: 2.43,
    },
  ],
  searched: {
    temp: 74,
    time: '12:41:09 PM',
    hi: 74,
    lo: 55,
    icon: '01d',
    condition: 'clear sky',
  },
  active: null,
  fetched: true,
};

export const JSON = {
  current: {
    dt: 1605739065,
    sunrise: 1605701583,
    sunset: 1605738778,
    temp: 58.5,
    feels_like: 51.89,
    pressure: 1031,
    humidity: 25,
    dew_point: 23.88,
    uvi: 3.4,
    clouds: 1,
    visibility: 10000,
    wind_speed: 3.36,
    wind_deg: 310,
    weather: [
      {
        id: 800,
        main: 'Clear',
        description: 'clear sky',
        icon: '01n',
      },
    ],
  },
  daily: [
    {
      dt: 1604851200,
      sunrise: 1604835818,
      sunset: 1604872819,
      temp: {
        day: 74.52,
        min: 55.87,
        max: 74.52,
        night: 61.5,
        eve: 72.34,
        morn: 56.86,
      },
      feels_like: {
        day: 73.72,
        night: 61.57,
        eve: 72.37,
        morn: 54.99,
      },
      pressure: 1028,
      humidity: 46,
      dew_point: 52.39,
      wind_speed: 2.73,
      wind_deg: 270,
      weather: [
        {
          id: 800,
          main: 'Clear',
          description: 'clear sky',
          icon: '01d',
        },
      ],
      clouds: 1,
      pop: 0,
      uvi: 2.98,
    },
    {
      dt: 1604937600,
      sunrise: 1604922286,
      sunset: 1604959164,
      temp: {
        day: 66.88,
        min: 57.09,
        max: 73.11,
        night: 60.24,
        eve: 71.01,
        morn: 58.06,
      },
      feels_like: {
        day: 67.14,
        night: 57.18,
        eve: 70.03,
        morn: 58.51,
      },
      pressure: 1028,
      humidity: 67,
      dew_point: 55.85,
      wind_speed: 2.64,
      wind_deg: 197,
      weather: [
        {
          id: 800,
          main: 'Clear',
          description: 'clear sky',
          icon: '01d',
        },
      ],
      clouds: 0,
      pop: 0,
      uvi: 3.12,
    },
  ],
};

const store = createStore(
  combineReducers(reducers),
  initialState,
  applyMiddleware(thunkMiddleWare)
);

export default store;
