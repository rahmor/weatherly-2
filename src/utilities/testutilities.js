import React from 'react';
import store from './testfixtures';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
export default function MyProvider(props) {
  const { children } = props;
  return <Provider store={store}>{children}</Provider>;
}

MyProvider.propTypes = {
  children: PropTypes.node,
};
