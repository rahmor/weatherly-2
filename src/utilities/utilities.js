import lodash from 'lodash';

function parseWeatherJSON(JSON) {
  let CURRENT = {
    temp: JSON.current.temp.toFixed(),
    time: parseUnixTime(JSON.current.dt),
    hi: JSON.daily[0].temp.max.toFixed(),
    lo: JSON.daily[0].temp.min.toFixed(),
    icon: JSON.current.weather[0].icon,
    condition: JSON.current.weather[0].description,
  };
  const DAILY = JSON.daily;
  return { CURRENT, DAILY };
}

function getCurrentFromDaily(JSON) {
  let CURRENT = {
    temp: JSON.temp.max.toFixed(),
    time: parseUnixTime(JSON.dt),
    hi: JSON.temp.max.toFixed(),
    lo: JSON.temp.min.toFixed(),
    icon: JSON.weather[0].icon,
    condition: JSON.weather[0].description,
  };
  return CURRENT;
}

function parseUnixTime(unix) {
  return new Date(unix * 1000).toLocaleTimeString('en-us');
}

const CITIESOBJECT = {
  'New York': [40.7127281, -74.0060152],
  Atlanta: [33.7490987, -84.3901849],
  'Los Angeles': [34.0536909, -118.2427666],
  Miami: [25.7741728, -80.19362],
  'San Francisco': [37.7790262, -122.4199061],
  Chicago: [41.8755616, -87.6244212],
  'Washington, DC': [38.8949855, -77.0365708],
  Boston: [42.3602534, -71.0582912],
};

const city = lodash.sample(Object.keys(CITIESOBJECT));
let coordinates = CITIESOBJECT[city];

export { coordinates, city, parseWeatherJSON, getCurrentFromDaily };
