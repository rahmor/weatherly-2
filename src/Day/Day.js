import React, { Component } from 'react';

import './Day.css';

const Day = ({ daily, i }) => {
  return (
    <div className='Day' key={i}>
      <h3>
        {new Date(daily.dt * 1000).toLocaleDateString('en-us', {
          weekday: 'long',
        })}
      </h3>
      {/* <p>Icon</p> */}
      {/* daily.weather[0].description */}
      <p>
        <span>Hi {daily.temp.max.toFixed()} &#176;</span> <span>&#183;</span>
        {'  '}
        <span>Lo {daily.temp.min.toFixed()} &#176;</span>
      </p>
    </div>
  );
};

export default Day;
