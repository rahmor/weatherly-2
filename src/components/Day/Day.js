import React from 'react';

import './Day.css';

const Day = ({ daily, i, focusDaily, updateActive, active }) => {
  return (
    <div
      className={`Day ${i === active ? 'active' : null}`}
      onClick={() => {
        focusDaily(daily);
        updateActive(i);
      }}
    >
      <h3>
        {new Date(daily.dt * 1000).toLocaleDateString('en-us', {
          weekday: 'long',
        })}
      </h3>
      {/* <p>Icon</p> */}
      {/* daily.weather[0].description */}
      <p>
        <span>Hi {daily.temp.max.toFixed()} &#176;</span> <span>&#183;</span>
        {'  '}
        <span>Lo {daily.temp.min.toFixed()} &#176;</span>
      </p>
    </div>
  );
};

export default Day;
