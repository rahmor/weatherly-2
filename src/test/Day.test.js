import React from 'react';
import Day from '../components/Day/Day';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { initialState } from '../utilities/testfixtures';

describe.only('<Day> component', () => {
  const props = {
    daily: initialState.daily[0],
    i: 1,
    focusDaily: jest.fn(),
    updateActive: jest.fn(),
    active: 1,
  };

  it('should match snapshot', () => {
    const wrapper = shallow(<Day {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('should call renderProps', () => {
    const wrapper = mount(<Day {...props} />);
    wrapper.find('div').at(0).simulate('click');
    expect(props.focusDaily).toHaveBeenCalled();
    expect(props.updateActive).toHaveBeenCalled();

    wrapper.unmount();
  });
});
