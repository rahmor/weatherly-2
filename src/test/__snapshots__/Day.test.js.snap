// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day> component should match snapshot 1`] = `
<div
  className="Day active"
  onClick={[Function]}
>
  <h3>
    Sunday
  </h3>
  <p>
    <span>
      Hi 
      75
       °
    </span>
     
    <span>
      ·
    </span>
      
    <span>
      Lo 
      56
       °
    </span>
  </p>
</div>
`;
